### ISSUE TITLE
Revoked access token returned during OAuth2 login 

### ISSUE THREAD

**Issue by Abilashini** [wso2/product-is#3515]
Product version - IS 5.1.0

### Steps to reproduce -

1. Create two OAuth based service providers SP1 and SP2

1. Login to SP1 and get the access token

1. Revoke the access token using the client credentials of SP2

1. Login to SP1 again

1. You will get the access token which you have revoked in step 3

### Reason - 
When we revoke the access token using the client credentials of SP2, the access token is getting revoked. But the cache of SP1 will be active with the old access token which will be returned during the login in step 4. 

### Potential improvement - 
Before revoking the access token [1], should check whether the access token belongs to the client which is trying to revoke. If it is a different client, then we should not allow the revocation. 

```
if (!accessTokenDO.getConsumerKey().equals(revokeRequestDTO.getConsumerKey())) {
                    revokeResponseDTO.setErrorCode(OAuth2ErrorCodes.INVALID_CLIENT);
                    revokeResponseDTO.setErrorMsg("Access token " + revokeRequestDTO.getToken() +" is not belong to the client with client ID " + revokeRequestDTO.getConsumerKey());
                    return revokeResponseDTO;
                }
```

[1] https://github.com/wso2-attic/carbon-identity/blob/master/components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java#L319

**Comment by emswbandara** [comment]
Fixed in IS-5.1.0. cannot reproduce in latest versions.

**Comment by madurangasiriwardena** [comment]
Result of https://wso2.org/jira/browse/IDENTITY-5375
